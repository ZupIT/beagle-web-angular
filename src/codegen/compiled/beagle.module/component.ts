/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import { Type } from '@angular/core'
import {
  kebabToCamelCase,
  removeExtraIndentation,
  replaceToUnderline,
} from '../../utils/formatting'
import { getComponentAnnotations } from '../../utils/metadata'
import { remoteViewSelector } from '../../../constants'


function createViewChildString(name: string, templateName: string, angularVersion: number): string {
  const componentName = replaceToUnderline(name)

  if (angularVersion >= 8) {
    return `'${componentName}': new ViewChild('${templateName}', { static: true })`
  } else {
    return `'${componentName}': new ViewChild('${templateName}')`
  }
}

function createQueries(components: Record<string, Type<any>>, angularVersion: number) {
  const componentNames = Object.keys(components)

  return componentNames.map((name) => {
    const selector = getComponentAnnotations(components[name]).selector
    const templateName = kebabToCamelCase(selector)
    return createViewChildString(name, templateName, angularVersion)
  })
}


export function createComponentString(
    components: Record<string, Type<any>>,
    angularVersion: number,
  ) {
  const queries = createQueries(components, angularVersion)

  // todo: legacy code. Remove the input "loadParams" with v2.0.
  const componentString = `
    @Component({
      selector: '${remoteViewSelector}',
      template,
      inputs: ['loadParams', 'route', 'networkOptions', 'controllerId'],
      queries: {
        ${queries.join(',\n        ')},
      },
    })
    export class BeagleRemoteView extends AbstractBeagleRemoteView {
      constructor(
        beagleProvider: BeagleProvider,
        ngZone: NgZone,
        changeDetector: ChangeDetectorRef,
      ) {
        // @ts-ignore
        super(beagleProvider, ngZone, changeDetector)
      }
    }
  `

  return removeExtraIndentation(componentString, 4)
}
