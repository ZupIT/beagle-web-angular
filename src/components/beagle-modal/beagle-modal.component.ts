/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import { Component, Input, ViewEncapsulation, ElementRef, OnInit, OnDestroy } from '@angular/core'
import { BeagleModalInterface } from '../schemas/modal'

@Component({
  selector: 'beagle-modal',
  templateUrl: './beagle-modal.component.html',
  styleUrls: ['./beagle-modal.component.less'],
  encapsulation: ViewEncapsulation.None,
})
export class BeagleModalComponent implements BeagleModalInterface, OnInit, OnDestroy {
  @Input() isOpen: boolean
  @Input() onClose: () => void

  constructor(private elementRef: ElementRef) {}

  ngOnInit() {
    document.addEventListener('keyup', (event) => this.closeOnEsc(event))
    this.elementRef.nativeElement.addEventListener('click', el => this.closeOnClick(el))
  }

  ngOnDestroy() {
    document.removeEventListener('keyup', (event) => this.closeOnEsc(event))
    this.elementRef.nativeElement.removeEventListener('click', el => this.closeOnClick(el))
  }

  handleClose() {
    this.onClose && this.onClose()
  }

  closeOnEsc({ key }: KeyboardEvent) {
    if (key === 'Escape') {
      this.handleClose()
      document.removeEventListener('keyup', this.handleClose)
    }
  }

  closeOnClick({ target }) {
    if (target.className === 'modal-background') {
      this.handleClose()
    }
  }
}
