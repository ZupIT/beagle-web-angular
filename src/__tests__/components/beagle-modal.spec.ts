/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import { TestBed, async, ComponentFixture } from '@angular/core/testing'
import { BeagleModalComponent } from '../../components/beagle-modal/beagle-modal.component'
import { buildBeagleTestModuleMetadata } from './test-module-builder'

let component: BeagleModalComponent
let fixture: ComponentFixture<BeagleModalComponent>


describe('BeagleModalComponent', () => {

  beforeEach(async(() => {
    TestBed.configureTestingModule(buildBeagleTestModuleMetadata([BeagleModalComponent]))
      .compileComponents()

    fixture = TestBed.createComponent(BeagleModalComponent)
    component = fixture.componentInstance
    component.onClose = jest.fn()
  }))

  it('should match snapshot', () => {
    expect(component).toMatchSnapshot()
  })

  it('should call onClose', () => {
    spyOn(component, 'handleClose').and.callThrough()
    component.handleClose()
    expect(component.onClose).toBeCalled()
  })

  it('should call onCLose on Esc event', () => {
    spyOn(component, 'closeOnEsc').and.callThrough()
    const event = new KeyboardEvent('event', { key: 'Escape' })
    component.closeOnEsc(event)
    expect(component.onClose).toBeCalled()
  })

  it('should call onClose on modal background click event', () => {
    spyOn(component, 'closeOnClick').and.callThrough()
    component.closeOnClick({ target: { className: 'modal-background' } })
    expect(component.onClose).toBeCalled()
  })


})
