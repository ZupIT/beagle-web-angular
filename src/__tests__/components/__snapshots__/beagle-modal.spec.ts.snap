// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeagleModalComponent should match snapshot 1`] = `
BeagleModalComponent {
  "elementRef": ElementRef {
    "nativeElement": <div
      id="root0"
    >
      <div
        class="modal"
      >
        <div
          class="modal-background"
        >
          <div
            class="modal-content"
          />
        </div>
      </div>
    </div>,
  },
  "onClose": [MockFunction],
}
`;
